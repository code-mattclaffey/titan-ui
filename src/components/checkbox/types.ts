import { GenericComponent } from '../../module';

export interface CheckboxInterface extends GenericComponent {
    id: string;
    name: string;
    required?: boolean;
    labelText: string;
    checked?: boolean;
    disabled?: boolean;
};