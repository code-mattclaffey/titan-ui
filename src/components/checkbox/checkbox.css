/* Checkbox Styling */

:root {
    --checkbox-indicator-size: 1em;
    --checkbox-indicator-border-radius: 2px;
    --checkbox-indicator-margin-right: 0.5em;
    --checkbox-indicator-top-position: 0.2em;
    --checkbox-indicator-border: var(--input-border);
    --checkbox-indicator-selected-size: 0.5em;
    --checkbox-indicator-selected-border: var(--input-border-focus);
    --checkbox-indicator-disabled-bg: var(--keyline);
    --checkbox-indicator-disabled-border-color: var(--keyline);
}

.e-checkbox {
    cursor: pointer;
    display: flex;
}

.e-checkbox__indicator {
    border: var(--input-border);
    border-radius: var(--checkbox-indicator-border-radius);
    flex: none;
    height: var(--checkbox-indicator-size);
    margin-right: var(--checkbox-indicator-margin-right);
    position: relative;
    top: var(--checkbox-indicator-top-position);
    width: var(--checkbox-indicator-size);
}

.e-checkbox__input:checked ~ .e-checkbox__indicator {
    border-color: var(--checkbox-indicator-selected-border);
}

.e-checkbox__input:disabled:checked ~ .e-checkbox__indicator,
.e-checkbox__input:disabled ~ .e-checkbox__indicator {
    background: var(--checkbox-indicator-disabled-bg);
    border-color: var(--checkbox-indicator-disabled-border-color);
}

.e-checkbox__input:checked ~ .e-checkbox__indicator::after {
    background: var(--primary);
    border-radius: var(--checkbox-indicator-border-radius);
    content: '';
    height: var(--checkbox-indicator-selected-size);
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: var(--checkbox-indicator-selected-size);
}

.e-checkbox__input:disabled:checked ~ .e-checkbox__indicator::after,
.e-checkbox__input:disabled ~ .e-checkbox__indicator::after {
    display: none;
}