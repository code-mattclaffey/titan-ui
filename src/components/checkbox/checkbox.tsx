import React from 'react';
import classNames from 'classnames';
import { CheckboxInterface } from './types';

export const Checkbox = ({
    additionalClassNames = '',
    name,
    id,
    checked = false,
    required = true,
    labelText,
    disabled,
}: CheckboxInterface) => {
    const classes = classNames('e-checkbox', additionalClassNames);

    return (
        <label htmlFor={id} className={classes}>
            <input
                type="checkbox"
                className="u-visually-hidden e-checkbox__input"
                name={name}
                id={id}
                required={required}
                defaultChecked={checked}
                disabled={disabled}
            />
            <span className="e-checkbox__indicator" />
            <span className="e-label">{labelText}</span>
        </label>
    );
};
