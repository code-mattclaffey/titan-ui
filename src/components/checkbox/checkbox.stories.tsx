import React from 'react';

import { Checkbox } from './checkbox';

export default {
    title: 'Checkbox',
    componentSubtitle: '',
};

export const checkbox = () => (
    <>
        <Checkbox
            name="hello-test"
            id="hello-test-checkbox-1"
            labelText="Checkbox One"
        />
        <Checkbox
            name="hello-test"
            id="hello-test-checkbox-box-2"
            labelText="Checkbox Two"
            checked
        />
        <Checkbox
            name="hello-test"
            id="hello-test-checkbox-box-3"
            labelText="Checkbox Three (Disabled)"
            disabled
        />
    </>
);
