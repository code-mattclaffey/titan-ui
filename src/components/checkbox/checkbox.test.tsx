import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Checkbox } from './checkbox';

afterEach(() => {
    cleanup();
});

describe('<Checkbox />', () => {
    it('should match snapshot in default state', () => {
        const { container } = render(
            <Checkbox
                name="hello-test"
                id="hello-test"
                labelText="Hello Test"
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot in checked state', () => {
        const { container } = render(
            <Checkbox
                name="hello-test"
                id="hello-test"
                labelText="Hello Test"
                checked
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot in disabled state', () => {
        const { container } = render(
            <Checkbox
                name="hello-test"
                id="hello-test"
                labelText="Hello Test"
                disabled
            />,
        );

        expect(container).toMatchSnapshot();
    });
});
