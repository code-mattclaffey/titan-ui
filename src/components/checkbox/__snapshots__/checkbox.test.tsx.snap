// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should match snapshot in checked state 1`] = `
<div>
  <label
    class="e-checkbox"
    for="hello-test"
  >
    <input
      checked=""
      class="u-visually-hidden e-checkbox__input"
      id="hello-test"
      name="hello-test"
      required=""
      type="checkbox"
    />
    <span
      class="e-checkbox__indicator"
    />
    <span
      class="e-label"
    >
      Hello Test
    </span>
  </label>
</div>
`;

exports[`<Checkbox /> should match snapshot in default state 1`] = `
<div>
  <label
    class="e-checkbox"
    for="hello-test"
  >
    <input
      class="u-visually-hidden e-checkbox__input"
      id="hello-test"
      name="hello-test"
      required=""
      type="checkbox"
    />
    <span
      class="e-checkbox__indicator"
    />
    <span
      class="e-label"
    >
      Hello Test
    </span>
  </label>
</div>
`;

exports[`<Checkbox /> should match snapshot in disabled state 1`] = `
<div>
  <label
    class="e-checkbox"
    for="hello-test"
  >
    <input
      class="u-visually-hidden e-checkbox__input"
      disabled=""
      id="hello-test"
      name="hello-test"
      required=""
      type="checkbox"
    />
    <span
      class="e-checkbox__indicator"
    />
    <span
      class="e-label"
    >
      Hello Test
    </span>
  </label>
</div>
`;
